### ISSUE TITLE
Automatically resolve scoped module slash in Windows when specified in transpile block

### ISSUE THREAD

**Issue by amoshydra** [nuxt/nuxt.js#4334]
### What problem does this feature solve?

Related to: https://github.com/nuxt/docs/pull/985

---

On Windows, the filepath that received by JavaScript uses backslash `\`.

At first it might be tempting to use
```js
build: {
  transpile: [
    '@scoped/packageA',
    '@scoped/packageB',
  ],
},
```

However, this will fail on Windows.

Related code:
```js
!modulesToTranspile.some(module => module.test(file))
```
https://github.com/nuxt/nuxt.js/blob/3702dfe5dcb5c4fdeb03fcd3025f5a5ae39232be/packages/webpack/src/config/base.js#L176

Which translate to:
```js
/@scoped\/packageA/.test('C:\\projects\\node_modules\\@scoped\\packageA\\src\\index.js')
```

and evaluated to: `false`

### What does the proposed changes look like?

We could convert filepath for user before testing, so that user can continue using one forward slash for scoped package.

```js
const normalizePathForRegex = module => path.join(module).replace(/\\/, '\\\\')
```

```js
  ...
  options.build.transpile = [].concat(options.build.transpile || [])
    .map(module => module instanceof RegExp ? module : new RegExp(normalizePathForRegex(module)));
  ...
)
```

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8142">#c8142</a>)</em></sub></div>

**Comment by amoshydra** [comment]
Digging throught the test file, it seems like, user are expected to write RegExp friendly string too

https://github.com/nuxt/nuxt.js/blob/dev/test/unit/basic.dev.test.js#L28-L31
```js
transpile: [
  'vue\\.test\\.js',
  /vue-test/
],
```

Perhaps, this shouldn't be the case too?

**Suggestion:**
User should be able to write normal string that represent a filepath and have their provided string sanitized for regex later.

In conjunction to the above feature request, if user provide a plain string, nuxt should
1. convert given path to os friendly file path
2. sanitize string for RegExp
3. convert the sanitized string to RegExp
4. use this RegExp for testing

**Comment by amoshydra** [comment]
Resolved by #4335
